econpizza/config.py

```diff
@@ -1,80 +1,53 @@
-"""Configuration object"""
-
 import os
 import jax
 
 class EconPizzaConfig(dict):
     def __init__(self, *args, **kwargs):
         super(EconPizzaConfig, self).__init__(*args, **kwargs)
-        self._enable_persistent_cache = False
-        self._econpizza_cache_folder = "__econpizza_cache__"
-        self._jax_cache_folder = "__jax_cache__"
-
-    @property
-    def enable_persistent_cache(self):
-        return self._enable_persistent_cache
-
-    @enable_persistent_cache.setter
-    def enable_persistent_cache(self, value):
-        self._enable_persistent_cache = value
-        self.setup_persistent_cache()
+        self.__dict__ = self
+        self.enable_persistent_cache = False
+        self.enable_jax_persistent_cache = False
+        self.jax_cache_folder = "__jax_cache__"
+        self.econpizza_cache_folder = "__econpizza_cache__"
+
+        self._setup_persistent_cache_map = {
+            # "enable_persistent_cache": self.setup_persistent_cache,
+            "enable_jax_persistent_cache": self.setup_persistent_cache_jax
+        }
 
-    @property
-    def jax_cache_folder(self):
-        return self._jax_cache_folder
+    def __setitem__(self, key, value):
+        return self.update(key, value)
     
-    @jax_cache_folder.setter
-    def jax_cache_folder(self, value):
-        self._jax_cache_folder = value
-    
-    @property
-    def econpizza_cache_folder(self):
-        return self._econpizza_cache_folder
-    
-    @econpizza_cache_folder.setter
-    def econpizza_cache_folder(self, value):
-        self._econpizza_cache_folder = value
-
     def update(self, key, value):
+        """Updates the attribute, and if it's related to caching, calls the appropriate setup function."""
         if hasattr(self, key):
             setattr(self, key, value)
+            if key in self._setup_persistent_cache_map and value:
+                self._setup_persistent_cache_map[key]()
         else:
             raise AttributeError(f"'EconPizzaConfig' object has no attribute '{key}'")
-    
+
     def _create_cache_dir(self, folder_name: str):
         cwd = os.getcwd()
         folder_path = os.path.join(cwd, folder_name)
         os.makedirs(folder_path, exist_ok=True)
-
         return folder_path
 
     def setup_persistent_cache(self):
```

Comment:
Thanks, looking good now! Last thing: this function only does econpizza cache things, which we agreed are for now always disabled. Could you remove this function?

Reply:
Certainly! Just to be clear: it's not called right now even if `enable_persistent_cache` is set to `True`.
So, my question is do I remove the function along with the tests for it (which are now skipped) and the flag itself (`enable_persistent_cache`)?

Reply:
I would say so. You can put it in the other pull request for later.

Reply:
Okay, done. Please have a quick look again and I think we can merge.

Reply:
Looks good. The mapping as a dict of functions is a nice idea!